src/m365/spo/commands/list/list-roleinheritance-reset.spec.ts

```diff
@@ -0,0 +1,189 @@
+      request.post
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.LIST_ROLEINHERITANCE_RESET), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('supports debug mode', () => {
+    const options = command.options();
+    let containsDebugOption = false;
+    options.forEach(o => {
+      if (o.option === '--debug') {
+        containsDebugOption = true;
+      }
+    });
+    assert(containsDebugOption);
+  });
+
+  it('supports specifying URL', () => {
+    const options = command.options();
+    let containsTypeOption = false;
+    options.forEach(o => {
+      if (o.option.indexOf('<webUrl>') > -1) {
+        containsTypeOption = true;
+      }
+    });
+    assert(containsTypeOption);
+  });
+
+  it('fails validation if both id and title options are not passed', () => {
+    const actual = command.validate({ options: { webUrl: 'https://contoso.sharepoint.com' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the url option is not a valid SharePoint site URL', () => {
+    const actual = command.validate({ options: { webUrl: 'foo' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if the url option is a valid SharePoint site URL', () => {
+    const actual = command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listId: '0CD891EF-AFCE-4E55-B836-FCE03286CCCF' } });
+    assert(actual);
+  });
+
+  it('fails validation if the id option is not a valid GUID', () => {
+    const actual = command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listId: '12345' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if the id option is a valid GUID', () => {
+    const actual = command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listId: '0CD891EF-AFCE-4E55-B836-FCE03286CCCF' } });
+    assert(actual);
+  });
+
+  it('fails validation if both id and title options are passed', () => {
+    const actual = command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listId: '0CD891EF-AFCE-4E55-B836-FCE03286CCCF', listTitle: 'Documents' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('reset role inheritance on list by title', (done) => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if ((opts.url as string).indexOf('/_api/web/lists/getbytitle(\'test\)/resetroleinheritance') > -1) {
+        return Promise.resolve();
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    command.action(logger, {
+      options: {
+        debug: true,
+        webUrl: 'https://contoso.sharepoint.com',
+        listTitle: 'test'
+      }
+    }, () => {
+      try {
+        assert(true);
```

Comment:
With this assertion this test always passes, no matter if the command actually works or not. Right now, we have an invalid URL in the test on line 113 and yet the test passes, because we're not checking if the command return an error. The correct way would to update line 126 to: `}, (err: any) => {` and then change this assert to: `assert.strictEqual(typeof err, 'undefined')`. In other words, if the command didn't return an error it succeeded.

Reply:
thanks for the detail explanation 👍

Reply:
done